main.c

```diff
@@ -819,33 +820,91 @@ resolve_all_solvables (Pool  *pool,
   return solv_failed;
 }
 
-static Queue
-pile_from_solvables (Pool  *pool,
-                     GStrv  solvables)
+static void
+add_solvable_to_pile (const char *solvable,
+                      Pool       *pool,
+                      Queue      *pile)
 {
-  Queue pile;
-  queue_init (&pile);
   int sel_flags = SELECTION_NAME | SELECTION_PROVIDES | SELECTION_GLOB |
                   SELECTION_CANON | SELECTION_DOTARCH;
```

Comment:
We could make `sel_flags` `static` so that it's initialized only once for the lifetime of the program, not on every call to `add_solvable_to_pile`.

Reply:
agree